pyomo/contrib/doe/measurements.py

```diff
@@ -44,14 +39,21 @@ def __init__(self, measurement_index_time, variance=None, ind_string='_index_'):
 
         Parameters
         ----------
-        measurement_index_time:
-            a ``dict``, keys are measurement variable names, 
-                if there are extra index, for e.g., Var[scenario, extra_index, time]:
-                    values are a dictionary, keys are its extra index, values are its measuring time points. 
-                if there are no extra index, for e.g., Var[scenario, time]:
-                    values are a list of measuring time point.
-            For e.g., for the kinetics illustrative example, it should be {'C':{'CA':[0,1,..], 'CB':[0,2,...]}, 'k':[0,4,..]},
-            so the measurements are C[scenario, 'CA', 0]..., k[scenario, 0]....
+        self_define_res: a ``list`` of ``string``, containing the measurement variable names with indexs, 
```

Comment:
This should be split into two functions instead of having all of these optional arguments. One function to directly specify the list of all measurement variable names. The section function to build up this list. That will make the documentation and examples much clearer. It will also be easier to test.